File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        int hLen = haystack.length();
        int nLen = needle.length();

        if (nLen > hLen) return -1;

        for (int i = 0; i <= hLen - nLen; i++) {
            boolean hasFound = true;

            for (int j = 0; j < nLen; j++) {
                if (haystack.charAt(i + j) != needle.charAt(j)) {
                    hasFound = false;
                    break;
                }
            }

            if (hasFound) return i;
        }

        return -1;
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.kt
class Solution {
    fun strStr(haystack: String, needle: String): Int {
        val hLen = haystack.length
        val nLen = needle.length

        if (nLen > hLen) return -1

        for (i in 0..hLen - nLen) {
            var hasFound = true

            for (j in 0 until nLen) {
                if (haystack[i + j] != needle[j]) {
                    hasFound = false
                    break
                }
            }

            if (hasFound) return i
        }

        return -1
    }
}